lib/google/serverless/exec.rb

```diff
@@ -813,6 +814,7 @@ def start_build_strategy app_info
           image = container ? container["image"] : image_from_build(app_info)
         else
           env_variables = {}
+          app_info["spec"]["template"]["spec"]["containers"][0]["env"].each { |env| env_variables[env["name"]] = env["value"] }
```

Comment:
Will this still work if there are no environment variables? (i.e. what if `app_info["spec"]["template"]["spec"]["containers"][0]` has no `"env"` key?)

Reply:
@dazuma No, it won't. I'm going to add an extra check here to account for that. A question I did have related to this was that the `"containers"`s shape is an array of hashes. Right now, I'm grabbing the environment variables via keying into the `"env"` field which is the first element in the `"containers"` array, but can we assume that no other fields will go before the `"env"` field in the `"containers"` array?

Reply:
So I think the structure of the big hash is determined by [this data structure](https://cloud.google.com/run/docs/reference/rpc/google.cloud.run.v1#google.cloud.run.v1.Service). Drilling down into the fields until you get to the "containers" field, [it says](https://cloud.google.com/run/docs/reference/rpc/google.cloud.run.v1#google.cloud.run.v1.RevisionSpec) "In Cloud Run, only a single container may be provided" which tells me "containers" is an array, and we're just picking up the first element (index 0), and that's fine because for cloud run there will probably be exactly one container. So I think we should be good.

Reply:
I see! That makes a lot more sense. `"containers"` will just be an array of size 1 for the Cloud Run case